In [24]:
import pandas as pd

# Лабораторная работа №5
## 1 Часть - Задание 2
### {Фамилия Имя}, {Номер группы}, Вариант {Номер варианта}, ({Дата})

## Задача и данные

Семь студентов проходят аттестацию по результатам теста и на основе оценки, поставленной им преподавателем. Результаты их аттестации таковы:

In [25]:
data = pd.DataFrame({
    ' (n)': [1, 2, 3, 4, 5, 6, 7],
    'Результаты теста (X)': [7, 2, 5, 4, 3, 8, 9],
    'Оценка преподавателя (Y)': [6, 3, 5, 7, 9, 10, 8]
})
data.set_index('Преподователи (n)', inplace=True)

n = len(data)

print(f'Количество наблюдений: {n}')

Количество наблюдений: 7


In [26]:
data.T

Преподователи (n)         1  2  3  4  5   6  7
Результаты теста (X)      7  2  5  4  3   8  9
Оценка преподавателя (Y)  6  3  5  7  9  10  8

На основе рангового коэффициента Спирмена найдите тесноту связи между двумя показателями оценки студента, рассматривая данные, как выборочные наблюдения случайных величин. Сделайте вывод

## Ранговый коэффициент Спирмена

### Присваиваем ранги

In [27]:
data['Ранг X'] = data['Результаты теста (X)'].rank()
data['Ранг Y'] = data['Оценка преподавателя (Y)'].rank()
pd.concat([data['Ранг X'], data['Ранг Y']], axis=1)

Ранг X  Ранг Y
Преподователи (n)                
1                     5.0     3.0
2                     1.0     1.0
3                     4.0     2.0
4                     3.0     4.0
5                     2.0     6.0
6                     6.0     7.0
7                     7.0     5.0

### Расчет коэффициента

$$p = 1 - \frac{6 \sum d_i^2}{n(n^2 - 1)}$$

In [28]:
d = data['Ранг X'] - data['Ранг Y']
p = 1 - (6 * sum(d ** 2)) / (n * (n ** 2 - 1))
print(f'Ранговый коэффициент Спирмена: {p:.3f}')

Ранговый коэффициент Спирмена: 0.464


Получаем коэффициент корреляции Спирмена равный $0.464$, что указывает на умеренную положительную связь между результатами теста и оценками преподавателя.

## Вывод

На основе рангового коэффициента Спирмена можно сделать вывод, что есть умеренная положительная связь между двумя показателями оценки студента. То есть, чем выше результаты теста, тем выше оценка преподавателя, и наоборот. Однако, необходимо учитывать, что выборка очень мала и не является репрезентативной для всей генеральной совокупности студентов.